packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -53,6 +54,32 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
       console.warn(`[${componentName}] \`${main}\` and \`${sub}\` cannot be used together. \`${sub}\` will be ignored.`)
     }
   }
+  const resolveTrailingSlashBehavior = (
+    to: RouteLocationRaw,
+    resolve: (to: RouteLocationRaw) => RouteLocation & { href?: string }
+  ): RouteLocationRaw | RouteLocation => {
+    if (options.trailingSlash !== 'append' && options.trailingSlash !== 'remove') {
+      return to
+    }
+
+    const forceAppend = options.trailingSlash === 'append'
+    if (typeof to === 'string') {
+      console.log('withTrailingSlash', to, withTrailingSlash(to, true))
+      return forceAppend ? withTrailingSlash(to, true) : withoutTrailingSlash(to, true)
+    } else if ('path' in to) {
+      return {
+        ...to,
+        path: forceAppend ? withTrailingSlash(to.path, true) : withoutTrailingSlash(to.path, true)
+      }
+    }
+
+    const route: RouteLocation = resolve(to)
+    return {
+      path: forceAppend ? withTrailingSlash(route.path, true) : withoutTrailingSlash(route.path, true),
+      query: route.query,
+      hash: route.hash
+    }
```

Comment:
this ends up limiting the resolved route to just those three properties - is there a reason?

Reply:
It will be later resolved in `RouterLink`. We should not provide `name` otherwise `path` will be overwritten. Others are not needed. These 3 are fully describing route AFAIK

Reply:
I think just route state/history data. I've pushed a fix but let me know if you think of a better approach.

Reply:
Looks good. I've just never seen a `to.state` usage in projects